build_tools/teatime.py

```diff
@@ -40,6 +40,81 @@
 import subprocess
 import sys
 import time
+import threading
+
+
+def periodic_log_sync(
+    log_path: Path, s3_bucket: str, s3_subdir: str, interval: int = 30
+):
+    def sync():
+        while True:
+            time.sleep(interval)
+            if not log_path.exists():
+                continue
+            try:
+                subprocess.run(
+                    [
+                        "aws",
+                        "s3",
+                        "cp",
+                        str(log_path),
+                        f"s3://{s3_bucket}/{s3_subdir}/{log_path.name}",
+                        "--content-type",
+                        "text/plain",
+                    ],
+                    check=True,
+                )
+            except Exception as e:
+                print(f"[teatime] periodic upload failed: {e}", file=sys.stderr)
+
+    thread = threading.Thread(target=sync, daemon=True)
+    thread.start()
+
+
+def upload_logs_to_s3(log_path: Path):
+    upload_to_s3 = os.getenv("TEATIME_S3_UPLOAD", "0") == "1"
+    s3_bucket = os.getenv("TEATIME_S3_BUCKET")
+    s3_subdir = os.getenv("TEATIME_S3_SUBDIR")
+    if not (upload_to_s3 and s3_bucket and s3_subdir):
+        return
+
+    logs_dir = log_path.parent.resolve()
+    try:
+        print(
+            f"[teatime] Uploading logs from {logs_dir} to s3://{s3_bucket}/{s3_subdir}/"
+        )
+        subprocess.run(
+            [
+                "aws",
+                "s3",
+                "cp",
+                str(logs_dir),
+                f"s3://{s3_bucket}/{s3_subdir}/",
+                "--recursive",
+                "--content-type=text/plain",
+                "--exclude",
+                "*",
+                "--include",
+                "*.log",
+            ],
+            check=True,
+        )
```

Comment:
Is this needed at all together with `periodic_log_sync`? As far as I understand this uploads, and therefore overwrites, all existing logs once called.